From: Robert Richter <robert.richter@amd.com>
Date: Wed, 9 Feb 2011 14:26:41 +0100
Subject: [PATCH 278/279] Fix redefinition of mark rodata compile error
Patch-mainline: no
References: FATE#311392, BNC#685313
Signed-off-by: Tony Jones <tonyj@suse.de>


Fix following compile error:

 init/main.c:95: error: redefinition of ‘mark_rodata_ro’
 include2/asm/cacheflush.h:194: note: previous definition of ‘mark_rodata_ro’ was here
 make[2]: *** [init/main.o] Error 1

Introduced with:

 commit 48954d1bdb7c39630dfb95448ba1570cf6452672
 Author: Nick Piggin <npiggin@suse.de>
 Date:   Wed Feb 10 01:19:27 2010 +1100

     - patches.suse/x86-mark_rodata_rw.patch: fix x86-64 bug (bnc#575194)
       (folded patch attached in that bug).

     suse-commit: 682b8ca5262a0433e60d9346d71e032f89a4c6fd

It is already defined in

 arch/x86/include/asm/cacheflush.h

Signed-off-by: Robert Richter <robert.richter@amd.com>
---
 arch/x86/include/asm/cacheflush.h |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/arch/x86/include/asm/cacheflush.h b/arch/x86/include/asm/cacheflush.h
index 2d94755..35e3627 100644
--- a/arch/x86/include/asm/cacheflush.h
+++ b/arch/x86/include/asm/cacheflush.h
@@ -191,7 +191,6 @@ extern int kernel_set_to_readonly;
 void set_kernel_text_rw(void);
 void set_kernel_text_ro(void);
 #else
-static inline void mark_rodata_ro(void) { }
 static inline void mark_rodata_rw(void) { }
 static inline void set_kernel_text_rw(void) { }
 static inline void set_kernel_text_ro(void) { }
-- 
1.7.3.4

